internal/js/modules/k6/browser/common/frame.go

```diff
@@ -1094,20 +1094,26 @@ func isQuotedText(s string) bool {
 	return false
 }
 
-// Locator creates and returns a new locator for this frame.
-func (f *Frame) GetByAltText(alt string, opts *GetByBaseOptions) *Locator {
-	f.log.Debugf("Frame:GetByAltText", "fid:%s furl:%q alt:%q opts:%+v", f.ID(), f.URL(), alt, opts)
-
-	a := "[alt=" + alt + "]"
-	if isQuotedText(alt) {
+// buildAttributeSelector is a helper method that builds an attribute selector
+// for use with the internal:attr engine. It handles quoted strings and
+// applies the appropriate suffix for exact or case-insensitive matching.
+func (f *Frame) buildAttributeSelector(attrName, attrValue string, opts *GetByBaseOptions) string {
```

Comment:
What is the expectation on this? If this used with high frequency, I expect we can use a strings builder with a bigger capacity and avoiding re-allocating the string in the case the execution joins the `isQuoted` branch.

If we need to apply it, then it would require a quick benchmark of `buildAttribtuteSelector` func.

Reply:
I'll work with strings builder, but not in this PR. I've added it as an action [item](https://github.com/grafana/k6/issues/4964) in a catch all issue that I'll work on once all the `getBy*` PRs are merged in.